src/huggingface_hub/inference/_client.py

```diff
@@ -763,6 +763,54 @@ def summarization(
         response = self.post(json=payload, model=model, task="summarization")
         return _bytes_to_dict(response)[0]["summary_text"]
 
+    def text_classification(
+        self, text: List[str], *, parameters: Optional[Dict[str, Any]] = None, model: Optional[str] = None
+    ) -> List[ClassificationOutput]:
```

Comment:
> For context, the philosophy behind `InferenceClient` is to simplify the user experience as much as possible - usually translating to a simplified but explicit interface. Worth case scenario, if the user really want a fully-featured interface, it's still possible to use the lower-level `InferenceClient().post(...)` method directly.
> 
> For this task, I would:
> 
> * accept only `str` as for the `text` input
> * remove `parameters` since [there isn't any relevant one](https://huggingface.co/docs/api-inference/detailed_parameters#text-classification-task) in the API.
> * return a `ClassificationOutput` instead of a list
> 
> This change would need you to slightly adapt the logic to still pass a list to the server and to extract the first item of the output.

Oh, I see! That makes sense.. I'll take a look at this and the other related PRs in the next few days.

Reply:
Ok, I think I understand what you want to achieve.
Similar to the `pipeline` (examples here: https://huggingface.co/tasks/text-classification), this should just return the highest-scoring classification.

One thing that confuses me about the `_bytes_to_dict` that I'm using from `huggingface_hub.inference._common` is that it states `def _bytes_to_dict(content: bytes) -> "Image":`... shouldn't that be `def _bytes_to_dict(content: bytes) -> "Dict":`? But if I perform the change I receive quite a few issues when I run `make quality`. Any advice?

Reply:
Wow, Good noticing this! It is completely a mistake from my side. The expected format should be either `Dict` or `List` but not `Image` of course. I made a fix in https://github.com/huggingface/huggingface_hub/pull/1641 to fix the related type checking issues. If you merge `main` into this branch, you should now be able to use `_bytes_to_dict` and `_bytes_to_list` normally. :)

Reply:
> Similar to the pipeline (examples here: https://huggingface.co/tasks/text-classification), this should just return the highest-scoring classification.

Hum no, let's return the list of all tags + score instead. So returning a `List[ClassificationOutput]` instead, as expected initially. My bad :confused:

Reply:
Ok, done. Apologies, I should have waited for your reply before performing the change and pushing a commit.

Should be all good now.